packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -284,6 +303,36 @@ class ShortcutManager extends ChangeNotifier with Diagnosticable {
     }
   }
 
+  /// Returns the intent, if any, that matches the current set of pressed keys.
+  Intent? matchingIntent({
+    KeySet<KeyboardKey>? keySet,
+  }) {
+    if (keySet == null && RawKeyboard.instance.keysPressed.isEmpty) {
+      return null;
+    }
+    keySet ??= LogicalKeySet.fromSet(RawKeyboard.instance.keysPressed);
+    Intent? matchedIntent = _shortcuts[keySet];
+    if (matchedIntent == null) {
+      // If there's not a more specific match, We also look for any keys that
+      // have synonyms in the map.  This is for things like left and right shift
+      // keys mapping to just the "shift" pseudo-key.
+      final Set<LogicalKeyboardKey> pseudoKeys = <LogicalKeyboardKey>{};
+      for (final KeyboardKey setKey in keySet.keys) {
+        if (setKey is LogicalKeyboardKey) {
+          final Set<LogicalKeyboardKey> synonyms = setKey.synonyms;
+          if (synonyms.isNotEmpty) {
+            // There currently aren't any synonyms that match more than one key.
```

Comment:
We should assert that that's actually true here so we notice when that actually ever changes.